code/game/machinery/constructable_frame.dm

```diff
@@ -1074,7 +1076,9 @@ to destroy them and players will be able to make replacements.
 							/obj/item/stack/sheet/glass = 1)
 
 /obj/item/circuitboard/merch
-	name = "Merchandise Computer Circuitboard"
+	board_name = "Merchandise Computer Circuit Board"
 	icon_state = "generic"
 	build_path = /obj/machinery/economy/merch
 	board_type = "machine"
+	req_components = list(
```

Comment:
Could use more components considering what other "vendor" machines have, maybe a matter bin? I dont know myself.

Reply:
All other vendor machines only have a restocking unit as their required component, but the Nanotrasen Merchandise Vendor isn't a normal "vendor" and has infinite stock as far as I can tell. I'd be willing to add a matter bin since it doesn't really change anything.